packages/protocol/src/peer/ControllerCommissioningFlow.ts

```diff
@@ -140,10 +140,53 @@ type CollectedCommissioningData = {
     fabricIndex?: FabricIndex;
 };
 
-/** Error that throws when Commissioning fails and process cannot be continued. */
-export class CommissioningError extends MatterError {}
+export enum CommissioningErrorCode {
```

Comment:
Is there a reason to use an enum here vs. creating a specialized version of CommissioningError for each different cause?

Would be nice to avoid creating secondary differentiation mechanism

One reason is that it would simplify error tests from e.g.

    `error instanceof CommissioningError && error.code === CommissioningErrorCode.CommissioningTimeout`

to

     `error instanceof CommissioningTimeoutError`

Reply:
I think it makes sense with StatusResponseError because we're using an enum defined by the spec. I still generally create specialized versions when I throw StatusResponseError too, often just having the derived class set the status code

You can still switch on the constructor with individual error classes 

In general I've kind of stopped using enums except when dealing with externally defined numeric identifiers... They're kind of unnatural in JS and add overhead of object lookup, and unnecessary in TS because you can use "union of strings" types

Reply:
My use case will be to catch the error and then send it back over my websocket channel for the client to handle.  I will have to create some type of mapping (numeric code, error name, etc...)  on both sides to handle this.  So an enum works nicely since i can easily map that, alternatively if you go the route of specialized error classes, it would be nice if they all conformed to an interface that has a "getErrorCode" or something like that, although i'm pretty flexible and will work with whatever is deemed best from a architecture POV.  

But i'm very excited about surfacing proper error codes, will make troubleshooting  so much easier!